src/libraries/Common/src/Interop/OSX/System.Security.Cryptography.Native.Apple/Interop.X509.macOS.cs

```diff
@@ -288,13 +290,25 @@ internal static partial class AppleCrypto
         {
             SafeSecIdentityHandle identityHandle;
             int osStatus;
+            int result;
 
-            int result = AppleCryptoNative_X509MoveToKeychain(
-                cert,
-                targetKeychain,
-                privateKey ?? SafeSecKeyRefHandle.InvalidHandle,
-                out identityHandle,
-                out osStatus);
+            // Ensure the keychain is not disposed, if there is any associated one
+            using (SafeKeychainHandle keychain = Interop.AppleCrypto.SecKeychainItemCopyKeychain(cert))
+            {
+                // AppleCryptoNative_X509MoveToKeychain can change the keychain of the input
+                // certificate, so we need to reflect that change.
+                SafeTemporaryKeychainHandle.UntrackItem(cert.DangerousGetHandle());
```

Comment:
Do we own the `cert` handle here? Can it get disposed of while it's in use?

What I'm really asking is: do we need to surround the `DangerousGetHandle` with an add/release ref?

Reply:
If you want to be pedantic, we should do it. Realistically this is only called with intermediate certificate that is not surfaced to the API user (AFAICT).

Reply:
> Realistically this is only called with intermediate certificate that is not surfaced to the API user (AFAICT).

That's more or less what I was asking - if another thread could dispose of the handle and we end up calling `UntrackItem(0)` or something. It's probably fine if we own the lifetime of the `SafeSecCertificateHandle` and don't expose it to user code.

Reply:
The only two places I see that get into her are SingleCert-PFX and Collection-PFX, and they're both done before the X509Certificate2 object (or the PAL object) leave the import function; so they're basically always locals.